src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,478 @@
+    else:
+        container_url = datasets_version
+        logging.info(f"Using user defined 'ncbi datasets' version '{container_url}'")
+
+    # Pull or load pre-existing 'datasets' singularity container image.
+    datasets_image = Client.pull(container_url, stream=False, pull_folder=image_dl_path, quiet=True)
+
+    return datasets_image
+
+
+def check_parameterization(input_cores: Path, input_accessions: Path, db_host: str, db_port: int) -> Path:
+    """Detect the kind of user input (cores/accessions) and determine any missing or
+    incorrect parameterization.
+
+    Args:
+        input_cores: input core(s) list file name.
+        input_accessions: input accession (s) list file name.
+        db_host: host server name
+        db_port: host serer port
+
+    Returns:
+        User input file used in assembly status querying
+    """
+    user_query_file: Path
+
+    # Input core names centered run
+    if input_cores:
+        user_query_file = input_cores
+        logging.info(f"Performing assembly status report using core db list file: {user_query_file}")
+        if db_host is None or db_port is None:
+            logging.critical(
+                "User must specify both arguments '--host' and '--port' when providing core database names."
+            )
+            sys.exit(1)
+    # Accession centered run
+    else:
+        user_query_file = input_accessions
+        logging.info(f"Performing assembly status report using INSDC accession list file: {user_query_file}")
+
+    return user_query_file
+
+
+def resolve_query_type(
+    query_list: list, partial_url: URL, input_cores: str, input_accessions: str
+) -> Union[Tuple[Dict, str]]:
+    """Function to identify the kind of queries being passed by user,
+    then extract the queries (core names or accessions) and store each with appropriate identifier.
+
+    Args:
+        query_list: List of user defined queries either core names, or accessions
+        partial_url: A partial MYSQL connection URL (host:port)
+        input_cores: arg parse param '--input_cores'
+        input_accessions: arg parse param '--input_accns'
+
+    Returns:
+        User queries stored as identifier[(core db name | UniqueID#)] : accession
+    """
+
+    query_accessions: Dict = {}
+    query_type: str = ""
+
+    if input_cores and input_accessions is None:
+        query_accessions = fetch_accessions_from_cores(query_list, partial_url)
+        query_type = "CoreDB"
+    elif input_cores is None and input_accessions:
+        query_type = "Accession"
+        for accession in query_list:
+            match = re.match(r"(GC[AF])_([0-9]{3})([0-9]{3})([0-9]{3})\.?([0-9]+)", accession)
+            if not match:
+                raise UnsupportedFormatError(f"Could not recognize GCA accession format: {accession}")
+            query_accessions[accession] = accession
+
+    return query_accessions, query_type
+
+
+def fetch_accessions_from_cores(database_names: list, connection_url: URL) -> Dict:
```

Comment:
To my understanding, `list` as a type hint will only work in Python 3.9+. You need to specify `List` for Python 3.8.